docs/source/daml-integration-kit/index.rst

```diff
@@ -0,0 +1,371 @@
+
+
+Architectural overview
+======================
+
+We explain how to implement a DAML ledger backed by a specific
+ledger `X`. This can be a proper distributed ledger or also just a database.
+The goal of this implementation is to allow multiple DAML applications, which are potentially run by
+different entities, to execute multi-party workflows using the ledger `X`. We
+illustrate this in the following diagram.
+
+.. image:: images/architecture-overview.svg
+
+.. original: https://www.lucidchart.com/invitations/accept/69799877-4e80-444d-96a3-3e90814e94df
+
+We assume that the `X` ledger allows entities to participate in the
+evolution of the ledger via particular nodes. In the remainder of this
+documentation, we call these nodes `participant nodes`.
+
+The boxes labeled `daml-on-<X>-server` denote the DAML Ledger API
+servers, which implement the DAML Ledger API on top of the services provided
+by the `X` participant nodes.
+
+The boxes labeled `<X>-daml-validator` denote `X`-specific DAML transaction
+validation services. In a distributed ledger they provide the ability for
+nodes to :doc:`validate DAML transactions </concepts/ledger-model/ledger-integrity>`
+at the appropriate stage in the `X` ledger's transaction commit process.
+Whether they are needed, by what nodes they are used, and whether they are run
+in-process or out-of-process depends on the `X` ledger's architecture. Above
+we depict a common case where the participant nodes jointly maintain the
+ledger's integrity and therefore need to validate DAML transactions.
+
+
+Message flow
+------------
+
+**TODO (BETA):**
+
+- explain to readers the life of a transaction at a high-level, so they have a
+  mental framework in place when looking at the example code.
+  (`GitHub issue <https://github.com/digital-asset/daml/issues/672>`__)
+
+
+Implementing your server and validator
+======================================
+
+Each `X` ledger requires at least the implementation of a specific
+`daml-on-<X>-server`. It might also require the implementation of a
+`<X>-daml-validator` as explained above. We provide two kinds of code to
+simplify their implementation.
+
+1. We provide Scala libraries for validating DAML transactions and serving the
+   Ledger API given implementations of two specific interfaces. See
+   :ref:`below <integration-kit_library_overview>` for an overview of these
+   libraries.
+
+2. We provide the `digital-asset/daml-on-x-example <https://github.com/digital-asset/daml-on-x-example>`__ GitHub repository,
+   which contains a complete example of a DAML Ledger backed by an in-memory
+   key-value store. It builds on our Scala libraries and demonstrates how they
+   can be assembled to serve the Ledger API and validate DAML transactions.
+
+   For ledgers where data is shared between all participant nodes, we
+   recommend using this example as a starting point for implementing your
+   server and validator. For ledgers with stronger privacy models, this example
```

Comment:
I would add a second paragraph break before "For ledgers with stronger privacy models"

Reply:
:+1: